src/librustc_resolve/lib.rs

```diff
@@ -611,7 +617,7 @@ struct UseError<'a> {
     /// Attach `use` statements for these candidates.
     candidates: Vec<ImportSuggestion>,
     /// The `NodeId` of the module to place the use-statements in.
-    node_id: NodeId,
+    def_id: DefId,
```

Comment:
```suggestion
    def_id: LocalDefId,
```
`NodeId`s are always local, so they can always be replaced with local `DefId`s (if they can be replaced with `DefId`s at all).

Reply:
So i tried but i don't _think_ its possible because data comes from `ModuleData::normal_ancestor_id` which must be a `DefId` can it can come from another crate.

Reply:
Hmm, the code previously used `unwrap()` to get a `NodeId` from `parent_scope.module.normal_ancestor_id`.
I agree that it's more robust to use a `DefId` here then given that `parent_scope` may point to other crates with macro 2.0 (I think).

`UsePlacementFinder` should still use a `LocalDefId`, it should just skip walking the crate entirely if the passed `target_module` is foreign.

Reply:
Ok i'll check if i can change `UsePlacementFinder` to use `LocalDefId`